core/src/main/scala/scala/pickling/tags/FastTypeTag.scala

```diff
@@ -0,0 +1,269 @@
+  def makeRaw[T](clazz: Class[T]): FastTypeTag[T] = {
+    rawclassToTagMap.getOrElse(clazz, {
+      if (clazz == null) FastTypeTag.Null
+      else if (clazz.isArray) {
+        val elemClass = clazz.getComponentType()
+        val elemTag = makeRaw(elemClass)
+        mkArrayTag(elemTag).asInstanceOf[FastTypeTag[T]]
+      } else {
+        val typeArgs = clazz.getTypeParameters
+        val clazzName0 = clazz.getName()
+        val clazzName =
+            if (clazzName0.endsWith("$")) clazzName0.replace("$", ".type")
+            else if (clazzName0.contains("anonfun$") || clazzName0.contains("$colon$colon") || clazzName0.endsWith("$sp")) clazzName0
+            else clazzName0.replace('$', '.')
+        if (typeArgs.isEmpty) SimpleFastTypeTag(clazzName, Nil)
+        else {
+          // For now just always use `Any`.  Ideally, we'd use some kind of placeholder.
+          val argTags = typeArgs map (i => Any)
+          FastTypeTag(clazzName, argTags.toList)
+        }
+      }
+   }).asInstanceOf[FastTypeTag[T]]
+  }
+
+  // Default tags
+  implicit val Any     = FastTypeTag[Any]("scala.Any")
+  implicit val Null    = FastTypeTag[Null]("scala.Null")
+  implicit val Byte    = FastTypeTag[Byte]("scala.Byte")
+  implicit val Short   = FastTypeTag[Short]("scala.Short")
+  implicit val Char    = FastTypeTag[Char]("scala.Char")
+  implicit val Int     = FastTypeTag[Int]("scala.Int")
+  implicit val Long    = FastTypeTag[Long]("scala.Long")
+  implicit val Boolean = FastTypeTag[Boolean]("scala.Boolean")
+  implicit val Float   = FastTypeTag[Float]("scala.Float")
+  implicit val Double  = FastTypeTag[Double]("scala.Double")
+  implicit val Unit    = FastTypeTag[Unit]("scala.Unit")
+  implicit val String  = FastTypeTag[java.lang.String]("java.lang.String")
+  implicit val AnyRef  = FastTypeTag[AnyRef]("scala.AnyRef")
+
+  // Arrays
+  private def mkArrayTag[T](el: FastTypeTag[T]): FastTypeTag[Array[T]] =
+    SimpleFastTypeTag("scala.Array", el :: Nil)
+  implicit val ArrayString: FastTypeTag[Array[String]] = mkArrayTag(String)
+  implicit val ArrayByte: FastTypeTag[Array[Byte]] = mkArrayTag(Byte)
+  implicit val ArrayShort: FastTypeTag[Array[Short]] = mkArrayTag(Short)
+  implicit val ArrayChar: FastTypeTag[Array[Char]] = mkArrayTag(Char)
+  implicit val ArrayInt: FastTypeTag[Array[Int]] = mkArrayTag(Int)
+  implicit val ArrayLong: FastTypeTag[Array[Long]] = mkArrayTag(Long)
+  implicit val ArrayBoolean: FastTypeTag[Array[Boolean]] = mkArrayTag(Boolean)
+  implicit val ArrayFloat: FastTypeTag[Array[Float]] = mkArrayTag(Float)
+  implicit val ArrayDouble: FastTypeTag[Array[Double]] = mkArrayTag(Double)
+  implicit val ArrayUnit: FastTypeTag[Array[Unit]] = mkArrayTag(Unit)
+  implicit val ArrayAnyRef: FastTypeTag[Array[AnyRef]] = mkArrayTag(AnyRef)
+
+  implicit val Nothing: FastTypeTag[Nothing] = 
+    FastTypeTag[Nothing]("scala.Nothing")
+  implicit val Ref = 
+    FastTypeTag[refs.Ref]("scala.pickling.refs.Ref")
```

Comment:
The `Ref` val does not have a type annotation. Also, I would not wrap those lines.